### ISSUE TITLE
Normalization

### ISSUE THREAD

**Issue by ryanxingql** [codeslake/DMENet#22]
Hi Lee,
I notice [here](https://github.com/codeslake/DMENet/blob/b9286625f045f22e7278512fcbca6b114e31ded1/main.py#L529) we store the estimated map after converting it into an image. This may introduce normalization.
I am wondering what is the network output range? What if the output ranges outside [0, 1]?
Thank you!

**Comment by codeslake** [comment]
Hi, @ryanxingql.

The predicted defocus map is in the range [[0, 1]](https://github.com/codeslake/DMENet/blob/b9286625f045f22e7278512fcbca6b114e31ded1/model.py#L225), so the [`scipy.misc.toimage()`](https://github.com/scipy/scipy/blob/v0.14.0/scipy/misc/pilutil.py#L261) function does not normalize the defocus map as we provided the appropriate range. Note that the ground truth defocus maps are also in [the same range between 0 and 1](https://github.com/codeslake/DMENet/blob/b9286625f045f22e7278512fcbca6b114e31ded1/utils.py#L43).

We've provided [`sigma_map`](https://github.com/codeslake/DMENet/blob/b9286625f045f22e7278512fcbca6b114e31ded1/main.py#L503) to get the actual sigma value.

**Comment by ryanxingql** [comment]
Thank you!
And also, what is the use of the sigma map? Is it for deblurring?

**Comment by codeslake** [comment]
> Thank you! And also, what is the use of the sigma map? Is it for deblurring?

Yes!

**Comment by ryanxingql** [comment]
Thank you so much ^ ^

**Comment by ryanxingql** [comment]
Hi @codeslake, I would like to ask you a question.

My understanding:
Our gray-scale defocus maps are ranging from 0 to 255.
Therefore, pixels with 0-values are focused, while the maximal available defocus value is 255 since the maximal COC for training is 15.
If a object in test images reaches out of this COC, its defocus value is still 255. In this situation, the output is somehow incorrect.

Am I right? Thank you.

**Comment by codeslake** [comment]
Yes, you are right.
DMENet will not be able to accurately predict the COC values of a test image, if the COC values are larger than the maximum COC used to create the training set.

**Comment by ryanxingql** [comment]
@codeslake Thank you!
